yt_dlp/extractor/twitch.py

```diff
@@ -525,6 +528,82 @@ def _extract_storyboard(self, item_id, storyboard_json_url, duration):
+        has_more_pages = True
+        retry_sleep = 5
+        max_retries = 3
+        retries = 0
+        pagenum = 1
+        gql_ops = [
+            {
+                'operationName': 'VideoCommentsByOffsetOrCursor',
+                'variables': {
+                    'videoID': vod_id,
+                    # 'cursor': <filled in in subsequent requests>
+                }
+            }
+        ]
+
+        self.to_screen('Downloading chat fragment pages')
+
+        while has_more_pages:
+            response = self._download_gql(vod_id, gql_ops, 'Downloading chat fragment page %d' % pagenum, fatal=False)
+
+            if response is False:
+                self.report_warning(f'Unable to fetch next chat history fragment. {retries + 1}. try of {max_retries}')
+
+                if retries < max_retries:
+                    retries += 1
+                    time.sleep(retry_sleep)
+                    continue
+                else:
+                    self.report_warning('Chat history download failed: retry limit reached')
+                    # TODO: when this happens, should I forget a partial chat history, or is it better to keep it?
+                    #       I think if I keep it, it might be better to persist a warning that it is incomplete
+                    # chat_history.clear()
+                    break
+
+            response_errors = traverse_obj(response, (..., 'errors'))
+            if response_errors is not None and len(response_errors) > 0:
+                self.report_warning(f"Error response recevied for fetching next chat history fragment: {response_errors}")
+
+            comments_obj = traverse_obj(response, (0, 'data', 'video', 'comments'))
+            chat_history.extend(traverse_obj(comments_obj, ('edges', ..., 'node')))
+
+            has_more_pages = traverse_obj(comments_obj, ('pageInfo', 'hasNextPage'))
+
+            if has_more_pages:
+                cursor = traverse_obj(comments_obj, ('edges', 0, 'cursor'))
+                if cursor is None:
+                    self.report_warning("Cannot continue downloading chat history: cursor is missing. There are additional chat pages to download.")
+                    break
+
+                pagenum += 1
+                gql_ops[0]['variables']['cursor'] = cursor
+
+            if has_more_pages is None:
+                cursor = traverse_obj(comments_obj, ('edges', 0, 'cursor'))
+
+                if cursor is not None:
+                    self.report_warning("Next page indication is missing, but found cursor. Continuing chat history download.")
+                else:  # In this case maintenance might be needed. Purpose is to prevent silent errors.
+                    self.report_warning("Next page indication is missing, and cursor not found.")
+
+        chat_history_length = len(chat_history)
+        self.to_screen('Extracted %d chat messages' % chat_history_length)
+        if chat_history_length == 0:
+            return None
+
```

Comment:
```suggestion

        if not chat_history:
            return
```

Reply:
Can I keep the `to_sceen()`? Maybe as a debug option, if it is possible. I like to look at it as kind of a verification; if numbers are too even I know something is not ok

Reply:
I don't think it's needed, but sure.
```py
        if not chat_history:
            return
        else:
            self.write_debug(f'Extracted {len(chat_history)} chat messages')
```